cmd/oras/copy.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/internal/trace"
+)
+
+type copyOptions struct {
+	option.Common
+	option.Remote
+
+	src        pullOptions
+	dst        pushOptions
+	rescursive bool
+	verbose    bool
+
+	debug bool
+}
+
+func copyCmd() *cobra.Command {
+	var opts copyOptions
+	cmd := &cobra.Command{
+		Use:     "copy <from-ref> <to-ref>",
+		Aliases: []string{"cp"},
+		Short:   "Copy files from ref to ref",
+		Long: `Copy artifacts from one reference to another reference
+	# Examples 
+	## Copy image only 
+	oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1
+	## Copy image and artifacts
+	oras cp -r localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1
+		`,
+		Args: cobra.MinimumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.src.targetRef = args[0]
+			opts.dst.targetRef = args[1]
+			return runCopy(opts)
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&opts.Remote.Configs, "from-config", nil, "source auth config path")
```

Comment:
I remember some options are already defined in the `common` and `remote`, do we need to specify them explicitly?

Reply:
`copy` is special because it takes 2 remote registry objects. @CharingJC Can you modify option.Remote to fit this change and remove defined cmd flags like `verbose` and `debug`?